pkg/services/ngalert/state/state.go

```diff
@@ -118,6 +121,8 @@ func (a *State) resultError(alertRule *ngModels.AlertRule, result eval.Result) {
 }
 
 func (a *State) resultNoData(alertRule *ngModels.AlertRule, result eval.Result) {
+	a.Error = result.Error
```

Comment:
The issue was originally reported and found in the `Alerting` state but I realised that this would also happen when the data is `noData`.